File: CallKit/CallKitObjCSample/CallKit/IncomingCallViewController.m
//
//  IncomingCallViewController.m
//  CallKit
//
//  Created by Dobrinka Tabakova on 9/26/16.
//  Copyright © 2016 Dobrinka Tabakova. All rights reserved.
//

#import "IncomingCallViewController.h"
#import "CallViewController.h"


static NSString *const kTitle = @"Incoming Call";

@interface IncomingCallViewController ()

@property (weak, nonatomic) IBOutlet UITextField *phoneNumberTextField;
@property (weak, nonatomic) IBOutlet UIButton *simulateCallButton;

@end


@implementation IncomingCallViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    self.title = kTitle;
}

#pragma mark - Actions

- (IBAction)phoneNumberValueChanged:(UITextField*)sender {
    NSString *phoneNumber = [sender.text stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceAndNewlineCharacterSet]];
    self.simulateCallButton.enabled = (phoneNumber.length);
}

- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
    if ([[segue identifier] isEqualToString:@"ShowCall"]) {
        CallViewController* callViewController = [segue destinationViewController];
        callViewController.phoneNumber = self.phoneNumberTextField.text;
        callViewController.isIncoming = YES;
        callViewController.uuid = [NSUUID new];
    }
}

@end

File: CallKit/CallKitSwiftSample/CallKitSwiftSample/IncomingCallViewController.swift
//
//  IncomingCallViewController.swift
//  CallKitSwiftSample
//
//  Created by Nikolay Andonov on 11/15/16.
//  Copyright © 2016 Mentormate. All rights reserved.
//

import UIKit

class IncomingCallViewController: UIViewController {
    
    private let kTitle = "Incomming Call"
    
    @IBOutlet weak var phoneNumberTextField: UITextField!
    @IBOutlet weak var simualteCallButton: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        title = kTitle
    }

    // MARK: - Actions
    @IBAction func phoneNumberValueChanged(_ sender: UITextField) {
        
        let phoneNumber = sender.text?.trimmingCharacters(in: .whitespacesAndNewlines)
        simualteCallButton.isEnabled = (phoneNumber?.characters.count)! > 0
    }

    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
        
        if segue.identifier == "ShowCall" {
            let callViewController = segue.destination as! CallViewController
            callViewController.phoneNumber = phoneNumberTextField.text
            callViewController.isIncoming = true
            callViewController.uuid = UUID.init()
        }
    }
}
